x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/SpatialCoordinateTypes.java

```diff
@@ -36,85 +34,23 @@ public long pointAsLong(double x, double y) {
         }
     },
     CARTESIAN {
-        public SpatialPoint longAsPoint(long encoded) {
-            try {
-                final double x = XYEncodingUtils.decode((int) (encoded >>> 32));
-                final double y = XYEncodingUtils.decode((int) (encoded & 0xFFFFFFFF));
-                return makePoint(x, y);
-            } catch (Error e) {
-                throw new IllegalArgumentException("Failed to convert invalid encoded value to cartesian point");
```

Comment:
I seem to remember that this exception wrapping was needed for some correct ESQL conversion of errors to null plus warnings. We needed explicitly to use IllegalArgumentException for that. Hopefully there are tests that will fail if this is not correctly done.

Reply:
If there is no test, there is no issue :)